lib/loader.js

```diff
@@ -35,6 +49,15 @@ function sassLoader(content) {
         throw new Error("Synchronous compilation is not supported anymore. See https://github.com/webpack-contrib/sass-loader/issues/333");
     }
 
+    if (asyncSassJobQueue == null) {
```

Comment:
Last note :smile: Seems we can avoid this block, just throw error in `catch` instead `loadingError = e;`

Reply:
Is it safe to throw an error at require-time? I figured you'd want them surfaced as webpack compiler errors -- I'm not sure how this works.

Reply:
@Timer Yep, it is not compilation error. It does not make sense because all attempts to compile `sass/scss` return error in callback. I think better throw error as fast as possible. @michael-ciniawsky what do you think about this?

Reply:
Wouldn't webpack's `bail` option cause it to stop on the first? I'm no expert here, just asking questions.

Reply:
I also think this block is redundant with the `try/catch` (L13+) and version check, but not 💯 😛  

```diff
- let loadingError = null;

try {
    const sass = require("node-sass");
    const sassVersion = require("node-sass/package.json").version;

    if (Number(sassVersion[0]) >= 4) {
        // This queue makes sure node-sass leaves one thread available for executing
        // fs tasks when running the custom importer code.
        // This can be removed as soon as node-sass implements a fix for this.
        const threadPoolSize = process.env.UV_THREADPOOL_SIZE || 4;

        asyncSassJobQueue = async.queue(sass.render, threadPoolSize - 1);
+       // Maybe not needed (see comment)       
+       if (asyncSassJobQueue === null) {
+           throw new Error("Please install `node-sass >= 4.x` to use `sass-loader`.")
+       }
    } else {
        throw new Error("The installed version of \`node-sass\` is not compatible (expected: >=4.x, actual: " + sassVersion + ").");
    }
} catch (err) {
+  // Catches 'both' (first thrown) {Error} thrown in the try block 
+   callback(err);

+   return;
}

...

-    if (asyncSassJobQueue == null) {
-        if (loadingError) {
-            callback(loadingError);
-            return;
-        }
-        callback(new Error("Please install `node-sass >=4.x` to use `sass-loader`."));
-        return;
-    }
```

But what's the exact reason for `asyncQueue === null`, in case `node-sass < 4` ? And isn't this basically a redundant (the same check) as `if (Number(sassVersion[0])) >= 4)` again ? Please elaborate what could possibly fail when queueing `sass.render` for thread pooling, since I'm not that familiar with `node-sass` :)

Reply:
Callback isn't available where you've added it, that's where the package itself is being evaluated.

AFAIK, it's generally poor practice to make a require throw an exception, but this might be normal practice in the webpack world.

The `loadingError` is to store the error until the plugin is used at compile time.

Reply:
Ooops... 😛you're right. What alternatives do we have to make this less confusing and more compact ?

Reply:
I'll come up with something.